aip/0216.md

```diff
@@ -60,13 +60,57 @@ The field referencing the `State` enum in a resource **should** behave and be
 documented as "Output only", in accordance with [AIP-203](./0203.md).
 
 APIs **should not** allow a `State` enum to be directly updated through an
-"update" method, and **should** instead use a custom method.
+"update" method, and **should** instead use a custom state transition method.
 
 This is because update methods are generally not expected to have side effects,
 and also because updating state directly implies that it is possible to set the
 state to any available value, whereas states generally reflect a resource's
 progression through a lifecycle.
 
+### State transition methods
+
+State transition methods should look like this:
+
+```proto
+// Publishes a book.
+// The `state` of the book after publishing will be PUBLISHED.
```

Comment:
Maybe add a comment about what states are valid before execution? And what happens if the book is already published?

Reply:
Done

Reply:
Incidentally, we should consider making this a proto annotation. This would be _really_ useful for something like a Kubernetes shim.